src/views/partials/settings.js

```diff
@@ -1,5 +1,183 @@
+  </button>
+`
+
+const createEmailItem = (email, breachCounts) => `
+  <li class='settings-email-item'>
+    <strong>
+      ${email.primary
+          ? `${getMessage('settings-email-label-primary', { email: email.email })}`
+          : email.email}
+    </strong>
+    ${email.verified
+        ? getMessage('settings-email-number-of-breaches-info', {
+            breachCount: breachCounts.get(email.email)
+          })
+        : emailNeedsVerificationSub(email)}
+
+    ${email.primary ? '' : deleteButton(email)}
+  </li>
+`
+
+// Moves the primary email to the front and sorts the rest alphabeticaly.
+const getSortedEmails = emails => [...emails].sort((a, b) => {
+  if (a.primary) {
+    return -1
+  }
+
+  if (b.primary) {
+    return 1
+  }
+
+  return a.email.localeCompare(b.email)
+})
+
+const createEmailList = (emails, breachCounts) => `
+  <ul class='settings-email-list'>
+    ${getSortedEmails(emails)
+      .map(email => createEmailItem(email, breachCounts))
+      .join('')}
+  </ul>
+`
+
+const optionInput = (csrfToken, { isChecked, option }) => `
+  <input
+    checked='${isChecked}'
+    class='js-settings-alert-options-input'
+    data-alert-option='${option}'
+    data-csrf-token='${csrfToken}'
+    name='settings-alert-options'
+    type='radio'
+  >
+`
+
+const alertOptions = csrfToken => `
+  <div class='settings-alert-options'>
+    <label class='settings-radio-input'>
+    ${optionInput(csrfToken, {
+      isChecked: true,
+      option: 0
+    })}
+    <span class='settings-radio-label'>
+      ${getMessage('settings-alert-preferences-option-one')}
+    </span>
+  </label>
+
+  <label class='settings-radio-input'>
+    ${optionInput(csrfToken, {
+      isChecked: false,
+      option: 1
+    })}
+    <span class='settings-radio-label'>
+      ${getMessage('settings-alert-preferences-option-two')}
+    </span>
+  </label>
+  </div>
+`
+
+const createAddEmailModal = limit => `
+  <dialog id='js-settings-modal' class='settings-email-modal'>
+    <button id='js-settings-close'>
+      <img src='/images/icon-close.svg'>
+    </button>
+    <img src='/images/settings-dialog-email.svg'>
+
+    <h3 class='settings-section-title'>
+      ${getMessage('settings-email-dialog-title')}
+    </h3>
+    <div id='js-settings-modal-content'>
+      ${getMessage('settings-email-limit-info', { limit })}
+      ${getMessage('settings-add-email-text')}
+    </div>
+
+    <div id='js-settings-modal-controls'>
+      <label>
+        ${getMessage('settings-email-input-label')}
+        <input id='js-settings-email-modal-input' type='text'>
+      </label>
+      <button id='js-settings-modal-send-verification' class='primary'>
+        ${getMessage('settings-send-email-verification-button')}
+      </button>
+    </div>
+  </dialog>
+`
+
+export const settings = data => {
+  const { breachCounts, csrfToken, emails, limit } = data
+
+  return `
+    <div id='js-settings' class='settings' data-csrf-token='${csrfToken}'>
```

Comment:
nit: since element IDs create global vars, should we avoid their use – at least on non-form elements?  If so, is this one avoidable?  The selector `.settings[data-csrf-token]` (or a similar variation) seems unique enough to me.

In the merged branch that I'm reviewing locally (prepared by @rhelmer) this ID is simply "settings", which is more of a concern as a global var.

Reply:
I agree on trying to avoid using `id`s. For entire pages I think it might be helpful to have something more descriptive than `.settings[data-csrf-token]`. I’ll add an identifier to the class list.